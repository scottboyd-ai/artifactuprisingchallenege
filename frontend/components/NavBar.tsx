import {
    Button,
    ClickAwayListener, createStyles, Drawer,
    IconButton, List, ListItem, ListItemIcon, ListItemText,
    makeStyles, Paper, Theme,
    Toolbar,
    Typography
} from "@material-ui/core";
import AppBar from "@material-ui/core/AppBar";
import {grey} from "@material-ui/core/colors";
import {Dashboard, ChevronLeft, ShoppingCart} from "@material-ui/icons";
import MenuIcon from '@material-ui/icons/Menu';
import clsx from 'clsx';
import * as React from "react";

const drawerWidth = 240;

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            display: 'flex',
            flexGrow: 1
        },
        title: {
            flexGrow: 1
        },
        appBar: {
            zIndex: theme.zIndex.drawer + 1,
            transition: theme.transitions.create(['width', 'margin'], {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen,
            }),
        },
        appBarShift: {
            marginLeft: drawerWidth,
            width: `calc(100% - ${drawerWidth}px)`,
            transition: theme.transitions.create(['width', 'margin'], {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen,
            }),
        },
        menuButton: {
            marginRight: '20px',
            backgroundColor: grey['50']
        },
        paper: {
            position: 'absolute',
            backgroundColor: "primary"
        },
        hide: {
            display: 'none',
        },
        drawer: {
            width: drawerWidth,
            flexShrink: 0,
            whiteSpace: 'nowrap',
        },
        drawerOpen: {
            width: drawerWidth,
            transition: theme.transitions.create('width', {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen,
            }),
        },
        drawerClose: {
            transition: theme.transitions.create('width', {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen,
            }),
            overflowX: 'hidden',
            width: theme.spacing(7) + 1,
            [theme.breakpoints.up('sm')]: {
                width: theme.spacing(9) + 1,
            },
        },
        toolbar: {
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'flex-end',
            padding: theme.spacing(0, 1),
            ...theme.mixins.toolbar,
        },
        content: {
            flexGrow: 1,
            padding: theme.spacing(3),
        },
    }),
);

export default function NavBar(props) {
    const classes = useStyles({});

    const [menuOpen, setMenuOpen] = React.useState(false);
    const [drawerOpen, setDrawerOpen] = React.useState(false);


    const handleMenuClick = () => {
        setMenuOpen((prev) => !prev);
    };

    const handleMenuClickAway = () => {
        setMenuOpen(false);
    };

    const handleDrawerOpen = () => {
        setDrawerOpen(true);
    };

    const handleDrawerClose = () => {
        setDrawerOpen(false);
    };

    return (
        <div className={classes.root}>
                <AppBar

                    position="fixed"
                    color="secondary"
                    className={clsx(classes.appBar, {
                        [classes.appBarShift]: drawerOpen,
                    })}
                >
                    <Toolbar>
                        <IconButton
                            color="inherit"
                            aria-label="open drawer"
                            onClick={handleDrawerOpen}
                            edge="start"
                            className={clsx(classes.menuButton, {
                                [classes.hide]: drawerOpen,
                            })}
                        >
                            <MenuIcon/>
                        </IconButton>
                        <Typography variant="h6" className={classes.title}>
                            Artifact Uprising
                        </Typography>
                        {/*Right Side*/}
                        <ClickAwayListener onClickAway={handleMenuClickAway}>
                            <div>
                                <IconButton
                                    aria-label="shopping cart"
                                    onClick={handleMenuClick}
                                    color={props.cart && props.cart.length ? "primary" : "secondary"}
                                >
                                    <ShoppingCart fontSize="large"/>
                                </IconButton>
                                {menuOpen ? (
                                    <Paper className={classes.paper}>
                                        <div><Button href="/checkout">Checkout</Button></div>
                                        <div><Button href="/clear">Clear Cart</Button></div>
                                    </Paper>
                                ) : null}
                            </div>
                        </ClickAwayListener>
                    </Toolbar>
                </AppBar>
                <Drawer
                    className={clsx(classes.drawer, {
                        [classes.drawerOpen]: drawerOpen,
                        [classes.drawerClose]: !drawerOpen,
                    })}
                    classes={{
                        paper: clsx({
                            [classes.drawerOpen]: drawerOpen,
                            [classes.drawerClose]: !drawerOpen,
                        }),
                    }}
                    open={drawerOpen}
                >
                    <div className={classes.toolbar}>
                        <IconButton onClick={handleDrawerClose}>
                            <ChevronLeft/>
                        </IconButton>
                    </div>
                    <List>
                        <ListItem key="Categories">
                            <ListItemIcon>
                                <Dashboard/>
                            </ListItemIcon>
                            <ListItemText primary="Categories" />
                        </ListItem>
                    </List>
                </Drawer>
                <main className={classes.content}>
                    <div className={classes.toolbar}/>
                    {props.children}
                </main>
        </div>
    );
}