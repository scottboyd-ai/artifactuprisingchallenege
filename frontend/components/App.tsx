import React from "react";
import $ from 'jquery';
import {Box, Button, Card, CardActions, CardContent, Grid, ThemeProvider, Typography} from "@material-ui/core";
import NavBar from "./NavBar";
import {mainTheme} from "../themes/mainTheme";

function App() {
    const [init, setInit] = React.useState(true);
    const [products, setProducts] = React.useState([]);
    const [cart, setCart] = React.useState([]);


    const getProducts = () => {
        $.ajax({
            url: '/products',
            type: 'GET',
            success: (response) => {
                setProducts(response);
            }
        })
    }

    const addItemToCart = (itemId) => {
        let newCart = [];
        if (!cart) {
            newCart.push(itemId);
        } else {
            newCart = [...cart, itemId];
        }
        setCart(newCart);
        saveCart(newCart);
    }

    const getCart = () => {
        $.ajax({
            url: '/cart',
            type: 'GET',
            success: (response) => {
                setCart(response);
            }
        })
    }

    const saveCart = (newCart) => {
        $.ajax({
            url: '/cart',
            type: 'POST',
            dataType: 'json',
            contentType: 'application/json; charset=utf-8',
            data: JSON.stringify({cart: newCart})
        })
    }

    const productsToItems = () => {
        if (products && products.length) {
            return products.map((item, i) => {
                return (
                    <Grid item key={i}>
                        <Card>
                            <CardContent>
                                <Typography>
                                    {item.name}
                                </Typography>
                                <Typography>
                                    ${item.price}
                                </Typography>
                            </CardContent>
                            <CardActions>
                                <Button size="small" onClick={() => {addItemToCart(item.id)}}>Add to cart</Button>
                            </CardActions>
                        </Card>
                    </Grid>
                )
            })
        }
    }

    if (init) {
        setInit(false);
        getProducts();
        getCart();
    }

    return (
        <ThemeProvider theme={mainTheme}>
            <NavBar cart={cart}>
                <Box>
                    <Grid container spacing={2}>
                        {productsToItems()}
                    </Grid>
                </Box>
            </NavBar>
        </ThemeProvider>
    );
}

export default App;