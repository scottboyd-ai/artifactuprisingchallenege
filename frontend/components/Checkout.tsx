import React from "react";
import {
    Button,
    Grid,
    Paper,
    Table,
    TableBody,
    TableCell,
    TableContainer,
    TableHead,
    TableRow,
    ThemeProvider
} from "@material-ui/core";
import {makeStyles} from "@material-ui/styles";
import $ from 'jquery';
import {mainTheme} from "../themes/mainTheme";
import {getCart, getProducts} from "../../services/RequestService";
import NavBar from "./NavBar";

const TAX_RATE = 0.07;

const useStyles = makeStyles({
    tableContainer: {
        maxWidth: 700
    },
    table: {
        minWidth: 700,
    },
});

export function Checkout() {
    const classes = useStyles();

    const [init, setInit] = React.useState(true);
    const [cart, setCart] = React.useState({});
    const [products, setProducts] = React.useState([]);
    const [cartCount, setCartCount] = React.useState(0);

    const getRows = () => {
        const rowArray = [];
        for (const product of products) {
            for (const [productId, cartItem] of Object.entries(cart)) {
                const cartItemObj: any = cartItem;
                if (product.id === productId) {
                    rowArray.push({
                        name: product.name,
                        quantity: cartItemObj.quantity,
                        price: product.price
                    })
                }
            }
        }
        return rowArray;
    }

    const getSum = (count, price) => {
        return currencyFormat(count * price);

    }

    const currencyFormat = (num: number) => {
        return `${num.toFixed(2)}`;
    }

    const subtotal = (items) => {
        return items.map(({ price, quantity }) => price * quantity).reduce((sum, i) => sum + i, 0);
    }

    const invoiceSubtotal = subtotal(getRows());
    const invoiceTaxes = TAX_RATE * invoiceSubtotal;
    const invoiceTotal = invoiceTaxes + invoiceSubtotal;

    const purchaseItems = () => {
        $.ajax({
            url: '/purchase',
            type: 'POST',
            success: (response) => {
                alert('Purchase made! Thank you for your patronage!');
                window.location.href = '/';
            }
        })
    }

    if (init) {
        setInit(false);
        getCart(setCart, setCartCount);
        getProducts(setProducts);
    }


    return (
        <ThemeProvider theme={mainTheme}>
            <NavBar setCart={setCart} cartCount={cartCount} setCartCount={setCartCount}>
                <Grid container>
                    <Grid item xs={3}/>
                    <Grid item xs={6}>
                        <Grid container justify="center">
                            <TableContainer component={Paper} className={classes.tableContainer}>
                                <Table className={classes.table} aria-label="spanning table">
                                    <TableHead>
                                        <TableRow>
                                            <TableCell align="center" colSpan={3}>
                                                Details
                                            </TableCell>
                                            <TableCell align="right">Price</TableCell>
                                        </TableRow>
                                        <TableRow>
                                            <TableCell>Desc</TableCell>
                                            <TableCell align="right">Qty.</TableCell>
                                            <TableCell align="right">Price</TableCell>
                                            <TableCell align="right">Sum</TableCell>
                                        </TableRow>
                                    </TableHead>
                                    <TableBody>
                                        {getRows().map((row) => (
                                            <TableRow key={row.name}>
                                                <TableCell>{row.name}</TableCell>
                                                <TableCell align="right">{row.quantity}</TableCell>
                                                <TableCell align="right">${row.price}</TableCell>
                                                <TableCell align="right">${getSum(row.quantity, row.price)}</TableCell>
                                            </TableRow>
                                        ))}
                                        <TableRow>
                                            <TableCell rowSpan={3}/>
                                            <TableCell colSpan={2}>Subtotal</TableCell>
                                            <TableCell align="right">${currencyFormat(invoiceSubtotal)}</TableCell>
                                        </TableRow>
                                        <TableRow>
                                            <TableCell>Tax</TableCell>
                                            <TableCell align="right">{`${(TAX_RATE * 100).toFixed(0)} %`}</TableCell>
                                            <TableCell align="right">${currencyFormat(invoiceTaxes)}</TableCell>
                                        </TableRow>
                                        <TableRow>
                                            <TableCell colSpan={2}>Total</TableCell>
                                            <TableCell align="right">${currencyFormat(invoiceTotal)}</TableCell>
                                        </TableRow>
                                    </TableBody>
                                </Table>
                            </TableContainer>
                        </Grid>
                    </Grid>
                    <Grid item xs={3}/>
                    <Grid item xs={4}/>
                    <Grid item xs={4}>
                        <Grid container justify="flex-end">
                            <Button href="/">Continue shopping</Button>
                            <Button onClick={purchaseItems}>Buy!</Button>
                        </Grid>
                    </Grid>
                    <Grid item xs={4}/>
                </Grid>
            </NavBar>
        </ThemeProvider>
    );
}