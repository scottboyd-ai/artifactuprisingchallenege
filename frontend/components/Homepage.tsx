import React from "react";
import {filterProducts, getCart, getProducts, saveCart} from "../../services/RequestService";
import {Box, Button, Card, CardActions, CardContent, Grid, ThemeProvider, Typography} from "@material-ui/core";
import {Rating} from "@material-ui/lab";
import {mainTheme} from "../themes/mainTheme";
import NavBar from "./NavBar";

export function Homepage() {
    const [init, setInit] = React.useState(true);
    const [products, setProducts] = React.useState([]);
    const [cart, setCart] = React.useState({});
    const [cartCount, setCartCount] = React.useState(0);

    const getProductById = (productId) => {
        return products.find((product) => {
            return product.id === productId;
        });
    }

    const getItemQuantity = (productId) => {
        const product = getProductById(productId);
        if (product) {
            return product.quantity;
        } else {
            return 0;
        }
    }

    const getCartQuantity = (productId) => {
        return cart && cart[productId] ? cart[productId].quantity : 0;
    }

    const addItemToCart = (productId) => {
        const product = getProductById(productId);
        if (product && product.quantity > 0) {
            let newCart: any = {};
            if (!cartCount) {
                newCart[productId] = {
                    quantity: 1
                };
            } else if (!cart[productId]) {
                newCart = {...cart};
                newCart[productId] = {
                    quantity: 1
                };
            } else {
                newCart = {...cart};
                newCart[productId] = {
                    quantity: cart[productId].quantity + 1
                };
            }
            setCart(newCart);
            const newCount = cartCount + 1;
            setCartCount(newCount);
            saveCart(newCart);
        }
    }

    const removeItemFromCart = (productId) => {
        const product = getProductById(productId);
        if (product) {
            let newCart: any = {};
            if (!cartCount) {
                return;
            } else if (cart && cart[productId]) {
                newCart = cart;
                const newProductCount = cart[productId].quantity - 1;
                if (newProductCount >= 0) {
                    newCart[productId] = {
                        quantity: cart[productId].quantity - 1
                    };
                } else {
                    delete newCart[productId];
                }
                setCart(newCart);
                const newCount = cartCount - 1;
                setCartCount( newCount);
                saveCart(newCart);
            }
        }
    }

    const addButtonDisabled = (productId) => {
        const originalProduct = getProductById(productId);
        let originalProductQuantity = 0;
        if (originalProduct) {
            originalProductQuantity = originalProduct.quantity;
        }
        const cartQuantity = cart && cart[productId] ? cart[productId].quantity : 0;

        return originalProductQuantity - cartQuantity <= 0;
    }

    const removeButtonDisabled = (productId) => {
        return (!cartCount || (!cart[productId] || (cart[productId] && cart[productId].quantity <= 0)));
    }

    const productsToItems = () => {
        if (products && products.length) {
            return products.map((product, i) => {
                return (
                    <Grid item key={i}>
                        <Card>
                            <CardContent>
                                <Typography variant="h6">
                                    {product.name}
                                </Typography>
                                <Typography>
                                    ${product.price}
                                </Typography>
                                <Rating value={product.rating} precision={0.1} readOnly size="small"/>
                                <Typography>
                                    {product.rating} / 5
                                </Typography>
                                <Typography>
                                    {getItemQuantity(product.id)} in stock
                                </Typography>
                                <Typography>
                                    {getCartQuantity(product.id)} in cart
                                </Typography>
                            </CardContent>
                            <CardActions>
                                <Button size="small" onClick={() => {addItemToCart(product.id)}} disabled={addButtonDisabled(product.id)}>Add to cart</Button>
                                <Button size="small" onClick={() => {removeItemFromCart(product.id)}} disabled={removeButtonDisabled(product.id)}>Remove from cart</Button>
                            </CardActions>
                        </Card>
                    </Grid>
                )
            })
        }
    }

    if (init) {
        setInit(false);
        getProducts(setProducts);
        getCart(setCart, setCartCount);
    }

    return (
        <ThemeProvider theme={mainTheme}>
            <NavBar setCart={setCart} cartCount={cartCount} setCartCount={setCartCount} filterProducts={filterProducts} setProducts={setProducts}>
                <Box>
                    <Grid container spacing={2}>
                        {productsToItems()}
                    </Grid>
                </Box>
            </NavBar>
        </ThemeProvider>
    );
}