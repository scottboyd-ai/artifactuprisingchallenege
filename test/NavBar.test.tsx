import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from "../frontend/components/NavBar";
import {Checkout} from "../frontend/components/Checkout";
import {Homepage} from "../frontend/components/Homepage";

describe('React', () => {
    test('NavBar renders', () => {
        const component = renderer.create(<NavBar/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Checkout renders', () => {
        const component = renderer.create(<Checkout/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Homepage renders', () => {
        const component = renderer.create(<Homepage/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});