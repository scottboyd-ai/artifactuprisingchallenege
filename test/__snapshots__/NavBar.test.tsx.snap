// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Checkout renders 1`] = `
<div
  className="makeStyles-root-137"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary makeStyles-appBar-139 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-141 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <h6
        className="MuiTypography-root makeStyles-title-138 MuiTypography-h6"
      >
        Artifact Uprising
      </h6>
      <div
        onClick={[Function]}
        onTouchEnd={[Function]}
      >
        <button
          aria-label="shopping cart"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-invisible"
              >
                0
              </span>
            </span>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
  <main
    className="makeStyles-content-148"
  >
    <div
      className="makeStyles-toolbar-147"
    />
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
        >
          <div
            className="MuiPaper-root MuiTableContainer-root makeStyles-tableContainer-135 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <table
              aria-label="spanning table"
              className="MuiTable-root makeStyles-table-136"
              role={null}
            >
              <thead
                className="MuiTableHead-root"
                role={null}
              >
                <tr
                  className="MuiTableRow-root MuiTableRow-head"
                  role={null}
                >
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                    colSpan={3}
                    scope="col"
                  >
                    Details
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    Price
                  </th>
                </tr>
                <tr
                  className="MuiTableRow-root MuiTableRow-head"
                  role={null}
                >
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    Desc
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    Qty.
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    Price
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
                    scope="col"
                  >
                    Sum
                  </th>
                </tr>
              </thead>
              <tbody
                className="MuiTableBody-root"
                role={null}
              >
                <tr
                  className="MuiTableRow-root"
                  role={null}
                >
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body"
                    rowSpan={3}
                  />
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body"
                    colSpan={2}
                  >
                    Subtotal
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    $
                    0.00
                  </td>
                </tr>
                <tr
                  className="MuiTableRow-root"
                  role={null}
                >
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body"
                  >
                    Tax
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    7 %
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    $
                    0.00
                  </td>
                </tr>
                <tr
                  className="MuiTableRow-root"
                  role={null}
                >
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body"
                    colSpan={2}
                  >
                    Total
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    $
                    0.00
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
        >
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            href="/"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <span
              className="MuiButton-label"
            >
              Continue shopping
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Buy!
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
      />
    </div>
  </main>
</div>
`;

exports[`React Homepage renders 1`] = `
<div
  className="makeStyles-root-137"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary makeStyles-appBar-139 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-141 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h6
        className="MuiTypography-root makeStyles-title-138 MuiTypography-h6"
      >
        Artifact Uprising
      </h6>
      <div
        onClick={[Function]}
        onTouchEnd={[Function]}
      >
        <button
          aria-label="shopping cart"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-invisible"
              >
                0
              </span>
            </span>
          </span>
        </button>
      </div>
    </div>
  </header>
  <main
    className="makeStyles-content-148"
  >
    <div
      className="makeStyles-toolbar-147"
    />
    <div
      className="MuiBox-root MuiBox-root-446"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      />
    </div>
  </main>
</div>
`;

exports[`React NavBar renders 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorSecondary makeStyles-appBar-3 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-5 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h6
        className="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
      >
        Artifact Uprising
      </h6>
      <div
        onClick={[Function]}
        onTouchEnd={[Function]}
      >
        <button
          aria-label="shopping cart"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <span
              className="MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <span
                className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-invisible"
              />
            </span>
          </span>
        </button>
      </div>
    </div>
  </header>
  <main
    className="makeStyles-content-12"
  >
    <div
      className="makeStyles-toolbar-11"
    />
  </main>
</div>
`;
